src/engine/engineconfig.cpp

```diff
@@ -104,10 +104,22 @@ std::vector<NamedInputMappingConfig> getDefaultMappings()
 
 void NamedInputMappingConfig::serialize(const serialization::Serializer<EngineConfig>& ser)
 {
-  ser(S_NV("name", name),
-      S_NV("controllerType", controllerType),
-      S_NV("gameMappings", gameMappings),
-      S_NV("menuMappings", menuMappings));
+  if(ser.loading)
+  {
+    ser(S_NV("name", name),
+        S_NV("controllerType", controllerType),
+        S_NVO("mappings", gameMappings),
+        S_NVO("gameMappings", gameMappings));
+
+    ser(S_NVD("menuMappings", menuMappings, getDefaultMappings().at(name == "Keyboard" ? 0 : 1).menuMappings));
```

Comment:
```suggestion
        S_NVO("gameMappings", gameMappings),
        S_NVD("menuMappings", menuMappings, getDefaultMappings().at(name == pgettext("Input|MappingName", "Keyboard") ? 0 : 1).menuMappings));
```

Also, please add a deprecation notice to `deprecations.txt`.

Reply:
This change causes the migration to no longer work. They need to be seperate calls to ser otherwise name isn't loaded yet at the point of evaluation.

For deprecations.txt is "1.14 remove menu input binding migration" sufficent?

Reply:
You're right, didn't pay attention enough to the data flow, still it's missing i18n for the name, which is important to work with other locales.

1.15 would be better for removal, this way the migration exists for 2 minor versions.